src/network/netdev/macvlan.c

```diff
@@ -51,6 +52,53 @@ static int netdev_macvlan_fill_message_create(NetDev *netdev, Link *link, sd_net
                         return log_netdev_error_errno(netdev, r, "Could not append IFLA_MACVLAN_MODE attribute: %m");
         }
 
+        if (m->bc_queue_length != UINT32_MAX) {
+                r = sd_netlink_message_append_u32(req, IFLA_MACVLAN_BC_QUEUE_LEN, m->bc_queue_length);
+                if (r < 0)
+                        return log_netdev_error_errno(netdev, r, "Could not append IFLA_MACVLAN_BC_QUEUE_LEN attribute: %m");
+        }
+
+        return 0;
+}
+
+int config_parse_macvlan_broadcast_queue_size(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        MacVlan *m = userdata;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+        assert(userdata);
+
+        if (isempty(rvalue)) {
+                m->bc_queue_length = UINT32_MAX;
+                return 0;
+        }
+
+        r = safe_atou32(rvalue, &m->bc_queue_length);
```

Comment:
Please do not assign the value to `m->bc_queue_length` before checking its validity.
```
uint32_t v;
(snip)
r = safe_atou32(rvalue, &v);
if (r < 0) {
    ...;
}
if (v == UINT32_MAX) {
   ..;
}

m->bc_queue_length = v;

return 0;
```

Reply:
By default `m->bc_queue_length = UINT32_MAX` . SO if safe_atou32 fails it still have that value it's not required to have a temp value.  Anyway UINT32_MAX not going to be assigned.